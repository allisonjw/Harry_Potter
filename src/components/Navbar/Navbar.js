import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import './Navbar.scss';
import { addSpells, addFavorite } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { PropTypes } from 'prop-types';

export const Navbar = props => {
  const { allSpells, favoriteCharacters, user, playerHouse} = props
  return (
    <div className='navbar'>
      <Link to='/login'>
      <img
      className={`nav_HP`}
      alt="harry potter intials"
      src={require('../../images/harry-potter-lightning-bolt-transparent-9.png')} 
      /></Link>
      <section className="user_info">
        <h2 className="nav_user-info">Welcome {user.name} the {user.wizardType} from {playerHouse}</h2>
      </section>
      <div className='nav_links'>
        <NavLink to='/CharmsContainer' onClick={() => addSpells(allSpells)}><button className='nav-btn'>Spells</button>
        </NavLink>
        <NavLink to='/FavoriteContainer' onClick={() => addFavorite(favoriteCharacters)}><button className='nav-btn'>Favorites</button>
        </NavLink>
      </div>
    </div>
  )
};

export const mapStateToProps = state => ({
  allSpells: state.allSpells,
  favoriteCharacters: state.favoriteCharacters,
  user: state.user,
  playerHouse: state.playerHouse
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ addSpells, addFavorite }, dispatch) 
};

export default connect(mapStateToProps)(Navbar);

Navbar.propTypes = {
  allSpells: PropTypes.array,
  favoriteCharacters: PropTypes.array
};
