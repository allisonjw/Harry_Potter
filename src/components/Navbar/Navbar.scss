.navbar {
    display: flex;
    background-color: #be7762;
    background-image: url("https://www.transparenttextures.com/patterns/brick-wall-dark.png");
    height: 100px;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 1%;
}

.nav_HP {
    color: black;
    width: 5em;
    margin: 7%;
    &:hover {
        cursor: pointer;
        transform: scale(1.1);
        transition: all .4s ease-in-out;
    }
}

.user_info {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
      .nav_user-info {
        color: white;
        text-shadow: 9px 9px 7px black;
        font-size: 2.5em;
        margin: 2%;
      }
}

.nav_links {
    display: flex;
    flex-direction: row;
    width: 30%;
    justify-content: space-evenly;
      .nav-btn {
          border-radius: 3px;
          box-shadow: 9px 9px 12px 0px black;
          font-size: 2em;
          height: 60px;
          &:hover {
            cursor: pointer;
            transform: scale(1.1);
            transition: all .4s ease-in-out;
        }
      }
}

