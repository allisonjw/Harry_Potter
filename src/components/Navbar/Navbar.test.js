import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps } from '../Navbar/Navbar';

describe('NavBar', () => {
    describe('Nav', () => {
      let mockUser = {
        name: 'name',
        wizardType: 'wizard',
        house: ''
      }

      let wrapper = shallow(<NavBar
          user={mockUser}
          allSpells={mockAllSpells}
          favoriteCharacters={mockFavoriteCharacters}
      />)

      it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('mapStateToProps', () => {
       it('should return user state', () => {
           const mockUserState = {
            name: 'name',
            wizardType: 'wizard',
            house: ''
          };
           
           const expected = {
            name: 'name',
            wizardType: 'wizard',
            house: ''
          };

           const mappedProps = mapStateToProps(mockUserState);
           expect(mappedProps).toEqual(expected);
       });
    });
});