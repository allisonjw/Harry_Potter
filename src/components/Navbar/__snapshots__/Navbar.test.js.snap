// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Nav should match snapshot with all information passing in correctly 1`] = `
<div
  className="navbar"
>
  <NavLink
    className="nav_h3"
    to="/login"
  >
    HP
  </NavLink>
  <section
    className="user_info"
  >
    <h3
      className="nav_user-info"
    >
      Welcome
    </h3>
    <h2
      className="nav_user-info"
    >
      name
    </h2>
    <h2
      className="nav_user-info"
    >
      the
    </h2>
    <h2
      className="nav_user-info"
    >
      wizard
    </h2>
  </section>
  <div
    className="nav_links"
  >
    <NavLink
      onClick={[Function]}
      to="/CharmsContainer"
    >
      <button
        className="nav-btn"
      >
        Spells
      </button>
    </NavLink>
    <NavLink
      onClick={[Function]}
      to="/FavoriteContainer"
    >
      <button
        className="nav-btn"
      >
        Favorites
      </button>
    </NavLink>
  </div>
</div>
`;
